File: web/remove_bg.py
import os
from PIL import Image

def remove_white_bg(img_path, out_path):
    img = Image.open(img_path)
    img = img.convert("RGBA")
    
    datas = img.getdata()
    newData = []
    
    # Tolerance for "white"
    threshold = 220
    
    for item in datas:
        # Check if the pixel is near white
        if item[0] > threshold and item[1] > threshold and item[2] > threshold:
            # Make it fully transparent
            newData.append((255, 255, 255, 0))
        else:
            # For non-white, we want to keep it, but to prevent white fringes,
            # we can make the opacity proportional to how dark it is (for anti-aliasing)
            # Or just keep it as is.
            # Simple approach:
            newData.append(item)
            
    img.putdata(newData)
    img.save(out_path, "PNG")

public_dir = r"d:\programming\AksharSetu\web\public"
files = ["books-left-ai.jpg", "books-right-ai.jpg"]

for f in files:
    in_path = os.path.join(public_dir, f)
    out_path = os.path.join(public_dir, f.replace(".jpg", ".png"))
    if os.path.exists(in_path):
        print(f"Processing {f}...")
        remove_white_bg(in_path, out_path)
        # Delete original jpg
        os.remove(in_path)
    else:
        print(f"File {f} not found.")

print("Done!")

File: web/remove_bg.js
const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

const publicDir = path.join(__dirname, 'public');
const files = ['doodle-smile', 'doodle-heart', 'doodle-star', 'doodle-bulb'];

async function processImages() {
  for (const name of files) {
    const inPath = path.join(publicDir, `${name}.jpg`);
    const outPath = path.join(publicDir, `${name}.png`);
    
    if (fs.existsSync(inPath)) {
      console.log(`Processing ${inPath}...`);
      try {
        // Read the image, extract alpha channel if we wanted, but we want to make near-white transparent.
        // The easiest way is to use a composite or just extract the raw buffer and modify it.
        const { data, info } = await sharp(inPath)
          .ensureAlpha()
          .raw()
          .toBuffer({ resolveWithObject: true });
          
        for (let i = 0; i < data.length; i += 4) {
          const r = data[i];
          const g = data[i+1];
          const b = data[i+2];
          // Near white
          if (r > 225 && g > 225 && b > 225) {
            data[i+3] = 0; // Set alpha to 0
          }
        }
        
        await sharp(data, {
          raw: {
            width: info.width,
            height: info.height,
            channels: 4
          }
        }).png().toFile(outPath);
        
        console.log(`Successfully converted ${name}.png`);
      } catch (e) {
        console.error(`Error processing ${name}:`, e);
      }
    }
  }
  console.log("All done!");
}

processImages();
